#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Debian glibc 2.3.1-10 specific regex bug fix patch.
# DP:		This applies posix/regcomp.c, cleaning allocated if buffer
# DP:		is cleaned up. Bug#175529 should be fixed.
# DP: Author: GOTO Masanori <gotom@debian.or.jp>
# DP: Upstream status: In CVS
# DP: Status Details: It's debian glibc 2.3.1-10 specific patch for fixing
# DP:		RC bugs.
# DP: Date: 2003-01-22

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
--- posix/regcomp.c.orig	2003-01-22 05:24:39.000000000 +0900
+++ posix/regcomp.c	2003-01-22 05:25:01.000000000 +0900
@@ -1,5 +1,5 @@
 /* Extended regular expression matching and search library.
-   Copyright (C) 2002 Free Software Foundation, Inc.
+   Copyright (C) 2002, 2003 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
    Contributed by Isamu Hasegawa <isamu@yamato.ibm.com>.
 
@@ -750,6 +750,7 @@
     {
       re_free (dfa);
       preg->buffer = NULL;
+      preg->allocated = 0;
       return err;
     }
 #ifdef DEBUG
@@ -763,6 +764,7 @@
     {
       re_free (dfa);
       preg->buffer = NULL;
+      preg->allocated = 0;
       return err;
     }
 
@@ -790,6 +792,7 @@
     re_compile_internal_free_return:
       free_dfa_content (dfa);
       preg->buffer = NULL;
+      preg->allocated = 0;
     }
 
   return err;
